localstack/config.py

```diff
@@ -491,32 +491,9 @@ def is_trace_logging_enabled():
 LOCALSTACK_HOSTNAME = os.environ.get("LOCALSTACK_HOSTNAME", "").strip() or LOCALHOST
 
 
-def parse_hostname_and_ip(
-    value: str, default_host: str, default_port: int = constants.DEFAULT_PORT_EDGE
-) -> str:
-    """
-    Given a string that should contain a <hostname>:<port>, if either are
-    absent then use the defaults.
-    """
-    host, port = default_host, default_port
-    if ":" in value:
-        hostname, port_s = value.split(":", 1)
-        if hostname.strip():
-            host = hostname.strip()
-        try:
-            port = int(port_s)
-        except (ValueError, TypeError):
-            pass
-    else:
-        if value.strip():
-            host = value.strip()
-
-    return f"{host}:{port}"
-
-
 def populate_legacy_edge_configuration(
     environment: Dict[str, str]
-) -> Tuple[str, str, str, int, int]:
+) -> Tuple["HostAndPort", List["HostAndPort"], str, int, int]:
```

Comment:
nit: It would not be necessary to use the forward reference for `HostAndPort` if you just move the class definition before this function.